src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.RemoteHostClientService.cs

```diff
@@ -248,7 +248,13 @@ private void OnConnectionChanged(object sender, bool connected)
 
                     // s_lastRemoteClientTask info should be saved in the dump
                     // report NFW when connection is closed unless it is proper shutdown
-                    FatalError.ReportWithoutCrash(new Exception("Connection to remote host closed"));
```

Comment:
Ah, I see.

I could see adding a new helper method on `FatalError` to simplify this. `FatalError.ReportWithoutCrash` was really always meant to be called in an exception filter; I don't really want to overload its meaning.

There are other changes I'm mulling over for how we report non-fatal exceptions, and such a helper would fit in better with those. So I'll keep it in mind.